public/js/index.js

```diff
@@ -0,0 +1,207 @@
+      $('#table-display-error').hide();
+      $('.table-display').empty();
+
+      for (let value of uploadedFiles) {
+        uploadedFileName = value.name;
+        readerForAllFiles(value, uploadedFileName);
+      }
+    } else if (fileToSearch.length === 0) {
+      $('#table-display-error').show();
+    } else {
+      $('#table-display-error').hide();
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        const data = e.target.result;
+        let invertedIndex = new InvertedIndex(JSON.parse(data));
+        const titleAndText = invertedIndex.getTitlesAndTexts();
+        const titleArray = getTitleArray(titleAndText);
+        const searchResult = invertedIndex.searchIndexedWords(wordToSearch);
+        displayTableTitle(titleArray, uploadedFileName, false);
+        displayTableBody(searchResult, uploadedFileName);
+      };
+
+      reader.readAsText(fileToSearch);
+    }
+  });
+
+  // back-upload clicked
+  $('#back-upload').click(() => {
+    $('.file-index').hide();
+    $('.file-upload').fadeIn('slow');
+  });
+});
+
+/*
+* readerForAllFiles (creates and read files)
+* @param {Object, string} value, uploadedFiles
+*/
+const readerForAllFiles = (value, uploadedFileName) => {
+  const wordToSearch = $('#search-field-id').val().split(' ');
+  const reader = new FileReader();
+  reader.readAsText(value);
+  reader.onload = (e) => {
+    const data = e.target.result;
+    const invertedIndex = new InvertedIndex(JSON.parse(data));
+    const titleAndText = invertedIndex.getTitlesAndTexts();
+    const titleArray = getTitleArray(titleAndText);
+    const searchResult = invertedIndex.searchIndexedWords(wordToSearch);
+    displayTableTitle(titleArray, uploadedFileName, true);
+    displayTableBody(searchResult, uploadedFileName);
+  };
+};
+
+/*
+* updateSelect (update the select option with the files selected)
+* @param {Object} uploadedFiles
+* @return{}
+*/
+const updateSelectOptions = (uploadedFiles, idToPopulate) => {
+  for (let arrayIndex = 0; arrayIndex < uploadedFiles.length; arrayIndex += 1) {
+    $(`<option id="option${arrayIndex + 1}" value="${uploadedFiles[arrayIndex].name}">${uploadedFiles[arrayIndex].name}</option>`)
+    .appendTo(idToPopulate);
+  }
+};
+
+/*
+* getSelectOptionFile (gets the file that will be worked on)
+* @param {String, Object} (fileName, uploadedFiles)
+* @return{Object}
+*/
+const getSelectOptionFile = (fileName, uploadedFiles) => {
+  for (let arrayIndex = 0; arrayIndex < uploadedFiles.length; arrayIndex += 1) {
+    if (fileName === uploadedFiles[arrayIndex].name) {
+      return uploadedFiles[arrayIndex];
+    }
+  }
+
+  return '';
+};
+
+/**
+ * getTitleArray (gets the titles in the array of titles and texts passed in as an argument)
+ * @param {array} titleTextArray
+ * @returns {array} - returns an array of the titles
+ */
+const getTitleArray = (titleTextArray) => {
+  const arrayToReturn = [''];
+
+  titleTextArray[0].titles.forEach((e) => {
```

Comment:
It'd be nice if you can  use a constant to index the `titleTextArray` instead of indexing it with the integer `0`. I understand that the `titles` are at the position 0, so you can have something like `titleTextArray[TitlesIndex]`; that way it's immediately obvious what value we're getting from the array.